### PR TITLE
Add 3D top bottom format support to VR video playing

### LINKED ISSUE #171: Fragmented subtitles in 3D side-by-side movie
问题1.电影片源是左右显示模式的，而字幕居中显示，然后在VR模式下，字幕明显割裂无法观看。
2.浏览器没法移动，我想躺着看，但是浏览器没有跟随眼睛移动的方式。
3.VR模式没有上下显示的模式，上下显示的电影片源无法使用。

### LINKED ISSUE #461: Add more options to view 3D-SBS-Videos
**Is your feature request related to a problem? Please describe.**
When watching videos in Wolvic, you can turn on 180 degree SBS projection. However there are no further options available to improve the viewing experience of these types of videos.

**Describe the solution you'd like**
I would like to see: (a) The option to zoom in or out of the 3D video. (b) Enable so-called "fish-eye" projection videos. (c) Enable 200, 220 and 360 degree videos.

### PR DESCRIPTION
**Pull request by HollowMan6**
Resolve #171

Test video: https://www.youtube.com/watch?v=9ScZRkRUYLM

Test video for YouTube auto format recognition: https://www.youtube.com/watch?v=CvxFiQYeHo0

### CODE REVIEW

**Review by javifernandez — APPROVED** [review]
The change looks good. I just wonder whether we should change the VIDEO_PROJECTION_3D_SIDE_BY_SIDE enum we already had with _LEFT_TO_RIGHT, in order to be a bit more consistent with the new enum we add and the 180_STEREO modes.

**Comment by HollowMan6** [comment]
> I just wonder whether we should change the VIDEO_PROJECTION_3D_SIDE_BY_SIDE enum we already had with _LEFT_TO_RIGHT, in order to be a bit more consistent with the new enum we add and the 180_STEREO modes.

I don't think we should, as 3D side-by-side and 3D top/bottom are what people usually call them.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java [hunk 1]
```diff
@@ -901,7 +901,8 @@ private void enterVRVideo(@VideoProjectionMenuWidget.VideoProjectionFlags int aP
         mAttachedWindow.setVisible(false);
 
         closeFloatingMenus();
-        if (aProjection != VideoProjectionMenuWidget.VIDEO_PROJECTION_3D_SIDE_BY_SIDE) {
+        if (aProjection != VideoProjectionMenuWidget.VIDEO_PROJECTION_3D_SIDE_BY_SIDE &&
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java:904]: I'd suggest refactoring this check in a new private method and use it in the three spots

**Review by svillar — APPROVED** [review]
Cool!

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/menus/VideoProjectionMenuWidget.java [hunk 2]
```diff
@@ -167,6 +172,8 @@ public void setSelectedProjection(@VideoProjectionFlags int aProjection) {
             return VIDEO_PROJECTION_180_STEREO_TOP_BOTTOM;
         } else if (projection.startsWith("180")) {
             return VIDEO_PROJECTION_180;
+        } else if (projection.startsWith("3dtb")) {
+            return VIDEO_PROJECTION_3D_TOP_BOTTOM;
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/menus/VideoProjectionMenuWidget.java:176]: Mind moving this change to the second commit?

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java (hunk 1 above)
Comment (HollowMan6) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java]: Done

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java [hunk 3]
```diff
@@ -1175,12 +1175,23 @@ public void onSharedPreferenceChanged(@NonNull SharedPreferences sharedPreferenc
         }
     }
 
+    private boolean isVRProjection(int projection){
+        switch (projection) {
+            case VIDEO_PROJECTION_NONE:
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java:1180]: This was not in the previous code so I wouldn't include it although I understand that the cases are similar (front facing videos).

Regarding `isVRProjection` I'd look for another name as 360 or 180 are also VR projections. Perhaps we should focus on the use case, something like `isFrontFacingVRProjection()`

**Review by svillar — CHANGES_REQUESTED** [review]

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java (hunk 3 above)
Comment (HollowMan6) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/NavigationBarWidget.java]: Done!

**Review by svillar — APPROVED** [review]

**Comment by svillar** [comment]
After landing I noticed that the logic is inverted. `isFrontFacingVRProjection()` should return true for both SBS and TB projections. Fixed it with a quick unreviewed commit.

**Pull request merged**